r/configure

```diff
@@ -244,6 +244,14 @@ if [ $? -eq 0 ]; then
     # NOTE: arrow-dataset is assumed to have the same -L flag as arrow
     # so there is no need to add its location to PKG_DIRS
   fi
+  # Check for Arrow Engine subcomponent
+  grep -i 'set(ARROW_ENGINE "ON")' $ARROW_OPTS_CMAKE >/dev/null 2>&1
+  if [ $? -eq 0 ]; then
+    PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_ENGINE"
+    PKG_LIBS="-larrow_engine $PKG_LIBS"
+    # NOTE: arrow-engine is assumed to have the same -L flag as arrow
+    # so there is no need to add its location to PKG_DIRS
+  fi
```

Comment:
In `configure.win`, could we add the following to configure_dev()?

```
  if [ $(cmake_option ARROW_ENGINE) -eq 1 ]; then
    PKG_CFLAGS="$PKG_CFLAGS -DARROW_R_WITH_ENGINE"
    PKG_CONFIG_PACKAGES="$PKG_CONFIG_PACKAGES arrow-engine"
  fi
```

(I wouldn't worry about `configure_release`, since there are corresponding CI changes that would need to be handled there).